xpumanager sidecar: add note about using HTTPS with xpum (#1738)
intel/intel-device-plugins-for-kubernetes: cmd/xpumanager_sidecar/README.md

  * [Install XPU-Manager with the Sidecar](#install-xpu-manager-with-the-sidecar)
  * [Install Sidecar to an Existing XPU-Manager](#install-sidecar-to-an-existing-xpu-manager)
* [Verify Sidecar Functionality](#verify-sidecar-functionality)

> * [Use HTTPS with XPU-Manager]()

Reviewer: XPU Manager project writes the name everywhere without '-': https://github.com/intel/xpumanager

so I think this doc should also write it like that.

And this entry should also have link to the corresponding section.

| -startup-delay | int | 10 | Startup delay before the first topology fetching (seconds, >= 0) |
| -label-namespace | string | gpu.intel.com | Namespace or prefix for the labels. i.e. **gpu.intel.com**/xe-links |

> | -allow-subdeviceless-links | bool | false | Include xelinks that are not on subdevices |

Reviewer: Name of the option indicates that the description should rather be:
```suggestion
| -allow-subdeviceless-links | bool | false | Include xelinks also for devices that do not have subdevices |
```

NOTE: The sidecar patch will remove other resources from the XPU-Manager container. If your XPU-Manager daemonset is using, for example, the smarter device manager resources, those will be removed.

### Verify Sidecar Functionality

You can verify the sidecar's functionality by checking node's xe-links labels:

```bash
$ kubectl get nodes -A -o=jsonpath="{range .items[*]}{.metadata.name},{.metadata.labels.gpu\.intel\.com\/xe-links}{'\n'}{end}"
master,0.0-1.0_0.1-1.1
```

### Use HTTPS with XPU-Manager

XPU-Manager can be configured to use HTTPS on the metrics interface. For the gunicorn sidecar, cert and key files have to be added to the command:
```
      - command:
        - gunicorn
...
        - --certfile=/certs/tls.crt

>         - --keyfile=/certs/tls.key

Reviewer: Add also example of mounting the secret volume(s) for these?

Author: I added it in the cert-manager section below.

### Verify Sidecar Functionality

You can verify the sidecar's functionality by checking node's xe-links labels:

```bash
$ kubectl get nodes -A -o=jsonpath="{range .items[*]}{.metadata.name},{.metadata.labels.gpu\.intel\.com\/xe-links}{'\n'}{end}"
master,0.0-1.0_0.1-1.1
```

### Use HTTPS with XPU-Manager

XPU-Manager can be configured to use HTTPS on the metrics interface. For the gunicorn sidecar, cert and key files have to be added to the command:
```
      - command:
        - gunicorn
...
        - --certfile=/certs/tls.crt
        - --keyfile=/certs/tls.key
...
        - xpum_rest_main:main()
```

The cert and key files have to be included in the container by a secret. Certificate and keys can be added manually or by leveraging [cert-manager](https://cert-manager.io/).

<details>
<summary>Example for the Cert-manager objects</summary>

```
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: selfsigned-issuer
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: serving-cert
spec:
  dnsNames:
  - xpum.svc
  - xpum.svc.cluster.local
  issuerRef:
    kind: Issuer
    name: selfsigned-issuer
  secretName: xpum-server-cert
```

> One would then include `xpum-server-cert` in the XPU-manager deployment.

Reviewer: As this part is not specific to using `cert-manager` for generating the secrets, I do not think it should be hidden under (same) `<details>`.

IMHO better would be to have it earlier, maybe in same example with `gunicorn` args:
```
    volumes:
    - name: certs
      secret:
        defaultMode: 420
        secretName: xpum-server-cert
...
    containers:
      - name: python-exporter
        volumeMounts:
        - mountPath: /certs
          name: certs
          readOnly: true
...
      - command:
        - gunicorn
...
        - --certfile=/certs/tls.crt
        - --keyfile=/certs/tls.key
...
        - xpum_rest_main:main()
```
